x-pack/plugins/observability_solution/apm/public/locator/onboarding_locator.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ObservabilityOnboardingLocatorParams } from '@kbn/deeplinks-observability';
+
+export enum ApmOnboardingLocatorCategory {
+  Apm = 'application',
+}
```

Comment:
Shouldn't this be managed by the onboarding team?

Reply:
They have [an issue](https://github.com/elastic/kibana/issues/189651) to enhance the locator. 

In the meantime, I thought it would be best for us to implement some restrictions on our side to prevent malformed URLs, even when using the locator.